src/pybind/mgr/dashboard/frontend/src/app/app.component.ts

```diff
@@ -1,18 +1,48 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { Title } from '@angular/platform-browser';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
 
 import { NgbPopoverConfig, NgbTooltipConfig } from '@ng-bootstrap/ng-bootstrap';
+import { filter, map } from 'rxjs/operators';
 
 @Component({
   selector: 'cd-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
-export class AppComponent {
-  constructor(popoverConfig: NgbPopoverConfig, tooltipConfig: NgbTooltipConfig) {
+export class AppComponent implements OnInit {
+  constructor(
+    popoverConfig: NgbPopoverConfig,
+    tooltipConfig: NgbTooltipConfig,
+    private titleService: Title,
+    private router: Router,
+    private activatedRoute: ActivatedRoute
+  ) {
     popoverConfig.autoClose = 'outside';
     popoverConfig.container = 'body';
     popoverConfig.placement = 'bottom';
 
     tooltipConfig.container = 'body';
   }
+
+  ngOnInit() {
+    const appTitle = this.titleService.getTitle();
+    this.router.events
+      .pipe(
+        filter((event) => event instanceof NavigationEnd),
+        map(() => {
+          let child = this.activatedRoute.firstChild;
+          while (child.firstChild) {
+            child = child.firstChild;
+          }
+          if (child.snapshot.data['breadcrumbs']) {
```

Comment:
@navinbarnwal This is a cleaner solution! But I found that is not working in all cases as not all routes have the breadcrumb set and also we're losing the full bread crumb ("Buckets" instead of "Ceph Object Gateway > Buckets"):
![Screenshot from 2021-06-11 15-07-06](https://user-images.githubusercontent.com/6780162/121697986-dd030080-cacd-11eb-9286-127e785596c3.png)

I suggest to replace all changes with this one in order to achieve same text as breadcrumb component:
![Screenshot from 2021-06-11 15-56-27](https://user-images.githubusercontent.com/6780162/121698112-fc9a2900-cacd-11eb-88d6-48fc87c16808.png)
```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.ts
index d933081ab9c..5727c269118 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.ts
@@ -23,6 +23,7 @@ THE SOFTWARE.
  */
 
 import { Component, Injector, OnDestroy } from '@angular/core';
+import { Title } from '@angular/platform-browser';
 import { ActivatedRouteSnapshot, NavigationEnd, NavigationStart, Router } from '@angular/router';
 
 import { concat, from, Observable, of, Subscription } from 'rxjs';
@@ -48,7 +49,7 @@ export class BreadcrumbsComponent implements OnDestroy {
   subscription: Subscription;
   private defaultResolver = new BreadcrumbsResolver();
 
-  constructor(private router: Router, private injector: Injector) {
+  constructor(private router: Router, private injector: Injector, private titleService: Title) {
     this.subscription = this.router.events
       .pipe(filter((x) => x instanceof NavigationStart))
       .subscribe(() => {
@@ -73,6 +74,14 @@ export class BreadcrumbsComponent implements OnDestroy {
           .subscribe((x) => {
             this.finished = true;
             this.crumbs = x;
+            this.titleService.setTitle(
+              `Ceph ${this.crumbs
+                .map((crumb: IBreadcrumb) => {
+                  return crumb.text.startsWith('http') ? '' : crumb.text;
+                })
+                .toString()
+                .replace(',', ' > ')}`
+            );
           });
       });
   }
```

Reply:
@navinbarnwal ping